Marlin/src/lcd/menu/menu_bed_corners.cpp

```diff
@@ -44,53 +44,179 @@
-  switch (bed_corner) {
-    case 0: current_position   = lf;   break; // copy xy
-    case 1: current_position.x = rb.x; break;
-    case 2: current_position.y = rb.y; break;
-    case 3: current_position.x = lf.x; break;
-    #if ENABLED(LEVEL_CENTER_TOO)
-      case 4: current_position.set(X_CENTER, Y_CENTER); break;
+#if ENABLED(LEVEL_CORNERS_USE_PROBE)
+
+  static inline void _lcd_level_bed_corners_probing() {
+    ui.goto_screen([]{ MenuItem_static::draw((LCD_HEIGHT - 1) / 2, GET_TEXT(MSG_PROBING_MESH)); });
+
+    float lfrb[4] = LEVEL_CORNERS_INSET_LFRB;
+    xy_pos_t lf { (X_MIN_BED) + lfrb[0] - probe.offset_xy.x , (Y_MIN_BED) + lfrb[1] - probe.offset_xy.y },
+             rb { (X_MAX_BED) - lfrb[2] - probe.offset_xy.x , (Y_MAX_BED) - lfrb[3] - probe.offset_xy.y };
+
+    do_blocking_move_to_z(LEVEL_CORNERS_Z_HOP - probe.offset.z);
+
+    switch (bed_corner) {
+      case 0: current_position   = lf;   break; // copy xy
+      case 1: current_position.x = rb.x; break;
+      case 2: current_position.y = rb.y; break;
+      case 3: current_position.x = lf.x; break;
+      #if ENABLED(LEVEL_CENTER_TOO)
+        case 4: current_position.set(X_CENTER - probe.offset_xy.x, Y_CENTER - probe.offset_xy.y); count_points_in_tolerance--; break;
+      #endif
+    }
+
+    do_blocking_move_to_xy(current_position);
+
+    #if ENABLED(BLTOUCH) && DISABLED(BLTOUCH_HS_MODE)
+      bltouch.deploy(); // DEPLOY in LOW SPEED MODE on every probe action
     #endif
+    TERN_(QUIET_PROBING, probe.set_probing_paused(true));
+
+    do_blocking_move_to_z(last_z - (LEVEL_CORNERS_PROBE_TOLERANCE), manual_feedrate_mm_s.z);// Move down until the probe is triggered
+    probe_triggered = TEST(endstops.trigger_state(), TERN(Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN, Z_MIN, Z_MIN_PROBE));// Check to see if the probe was triggered
+
+    if (!probe_triggered) {
+      ui.goto_screen([]{
+        MenuItem_confirm::select_screen(
+          GET_TEXT(MSG_BUTTON_DONE), GET_TEXT(MSG_BUTTON_SKIP)
+          , []{ corner_probing_done = true;
+                wait_for_probe = false;
+                TERN_(HAS_LEVELING, set_bed_leveling_enabled(leveling_was_active));
+                ui.goto_previous_screen_no_defer();
+            }
+          , []{ wait_for_probe = false; }
+          , GET_TEXT(MSG_LEVEL_CORNERS_RAISE)
+          , (const char*)nullptr, PSTR("")
+        );
+      });
+      ui.set_selection(true);
+
+      wait_for_probe = true;
+      while (wait_for_probe && !probe_triggered) {
+        probe_triggered = TEST_PROBE_PIN;
+        idle();
+      }
+      wait_for_probe = false;
+    }
+
+    TERN_(QUIET_PROBING, probe.set_probing_paused(false));
+
+    #if ENABLED(BLTOUCH) && DISABLED(BLTOUCH_HS_MODE)
+      bltouch.stow();
+    #endif
+
+    if (probe_triggered) {
+      endstops.hit_on_purpose();
+      if (!WITHIN(current_position.z, current_position.z - (LEVEL_CORNERS_PROBE_TOLERANCE), current_position.z + (LEVEL_CORNERS_PROBE_TOLERANCE))) {
```

Comment:
This will always evaluate the same way, because always (A-B) <= A <= (A+B).

Reply:
I'm not sure. The ideia is to check if current_position.z if not (!) between the tolerance range.

According to vscode:
#define WITHIN(N,L,H) ((N) >= (L) && (N) <= (H))
Expands to:
((current_position.z) >= (current_position.z - 0.1) && (current_position.z) <= (current_position.z + 0.1))

Maybe I'm wrong, but it seens fine to me

Reply:
It will always evaluate `TRUE` regardless of the value of `current_position.z`.
It is always true that `z-p <= z <= z+q` when `p` and `q` are positive or zero.

Reply:
Maybe you want to check whether a possible new Z is in range before assigning it to `current_position.z`.

Reply:
Now I got it, thank you